### DISCUSSION TITLE
define what services are doing exactly

### DISCUSSION THREAD

**Discussion by fredfred41** [mcguirepr89/BirdNET-Pi#600] [Q&A] [answered]
Hi, I'm trying to install to PI 3 model B+, I followed the specific "RPi0W2 Installation Guide", and I would like to shut down some services to improve the speed and stability.
But I don't really understand the real function of every services...
Extraction Service, BirdNET Analysis Server, BirdNET Analysis Client, specifically.
Is there somewhere an explanation more detailled, or could someone explain this here?
Thanks a lot.

**Comment by mcguirepr89** [comment, marked as answer]
Hi, @fredfred41 

You will want to always have those three services as well as the recording service stay active as you will understand from their descriptions below:
- `BirdNET Analysis Server`: This service is responsible for building the classification model and loading its corresponding species labels. These tasks have been placed into this service as distinct from the `BirdNET Analysis Client` so that the system only builds the model and loads the labels once, while the _`Client`_ service below supplies the data to be analyzed. It also populates the database and sends out any configured notifications.
- `BirdNET Analysis Client`: As mentioned above, this service supplies the analysis server with the data that should be analyzed, i.e., the audio and its meta data.
- `Extraction Service`: This service parses the output from the species classification performed from the services above and extracts the identified birdsounds into their own audio clips while also creating a spectrogram for the extraction.

I hope that is helpful!

**Reply by fredfred41** [reply]
thanks a lot, 🎵 I can see clearly now 🎵
